src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/CrossTarget/TargetFrameworkProvider.cs

```diff
@@ -3,71 +3,32 @@
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
 using System.Linq;
-using Microsoft.VisualStudio.ComponentModelHost;
-using Microsoft.VisualStudio.Shell;
 using NuGet.VisualStudio;
 
 namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.CrossTarget
 {
     [Export(typeof(ITargetFrameworkProvider))]
     [AppliesTo(ProjectCapability.DependenciesTree)]
-    internal class TargetFrameworkProvider : OnceInitializedOnceDisposed, ITargetFrameworkProvider
+    internal sealed class TargetFrameworkProvider : ITargetFrameworkProvider
     {
         [ImportingConstructor]
-        public TargetFrameworkProvider(SVsServiceProvider serviceProvider)
+        public TargetFrameworkProvider(
+            IVsFrameworkCompatibility nugetComparer,
+            IVsFrameworkParser nugetFrameworkParser)
         {
-            ServiceProvider = serviceProvider;
+            NugetComparer = nugetComparer;
+            NugetFrameworkParser = nugetFrameworkParser;
         }
-        
-        private SVsServiceProvider ServiceProvider { get; }
-        protected IVsFrameworkCompatibility NugetComparer { get; set; }
-        protected IVsFrameworkParser NugetFrameworkParser { get; set; }
 
-        private readonly object _targetsLock = new object();
-        private List<ITargetFramework> CachedTargetFrameworks = new List<ITargetFramework>();
-
-        protected override void Initialize()
-        {
-            if (!ShouldInitialize())
-            {
-                return;
-            }
-
-            ThreadHelper.JoinableTaskFactory.Run(async delegate
-            {
-                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
-
-                var container = ServiceProvider.GetService<IComponentModel, SComponentModel>();
-                if (container != null)
-                {
-                    NugetComparer = container.GetExtensions<IVsFrameworkCompatibility>().FirstOrDefault();
-                    NugetFrameworkParser = container.GetExtensions<IVsFrameworkParser>().FirstOrDefault();
-                }
-            });
-        }
+        private IVsFrameworkCompatibility NugetComparer { get; }
 
-        /// <summary>
-        /// For unit tests to avoid UI initialization.
-        /// </summary>
-        /// <returns></returns>
-        protected virtual bool ShouldInitialize()
-        {
-            return NugetComparer == null || NugetFrameworkParser == null;
-        }
+        private IVsFrameworkParser NugetFrameworkParser { get; }
 
-        protected override void Dispose(bool disposing)
-        {
-            if (disposing)
-            {
-                NugetComparer = null;
-                NugetFrameworkParser = null;
-            }
-        }
+        private readonly object _targetsLock = new object();
+        private List<ITargetFramework> CachedTargetFrameworks = new List<ITargetFramework>();
```

Comment:
2 questions about `CachedTargetFrameworks`

1. I believe we are using the default string comparer below, don't we want to use `OrdinalIgnoreCase`?
```C#
targetFramework = CachedTargetFrameworks.FirstOrDefault(x => x.Equals(shortOrFullName));
```
2.  below adds a target framework to the cache if the `ITargetFramework` does not match our `shortOrFullName`.  Is there ever a case where we expect two different `ITargetFramework`s to resolve to the same string?  If not, couldn't this be a dictionary instead of a list?
```C#
var frameworkName = NugetFrameworkParser.ParseFrameworkName(shortOrFullName);
if (frameworkName != null)
{
    var shortName = NugetFrameworkParser.GetShortFrameworkName(frameworkName);
    targetFramework = new TargetFramework(frameworkName, shortName);
    // remember target framework - there can not bee too many of them across the solution.
    CachedTargetFrameworks.Add(targetFramework);
}
```